arkouda/series.py

```diff
@@ -192,31 +193,29 @@ def isin(self, lst):
         boolean = in1d(self.values, lst)
         return Series(data=boolean, index=self.index)
 
-    def locate(self, key):
+    @typechecked
+    def locate(self, key: Union[int, pdarray, Series, List, Tuple]) -> Series:
         """Lookup values by index label
 
-        The input can be a scalar, a list of scalers, or a list of lists (if the series has
-        a MultiIndex).
-        As a special case, if a Series is used as the key, the series labels are preserved
-        with its values used as the key.
+        The input can be a scalar, a list of scalers, or a list of lists (if the series has a
+        MultiIndex). As a special case, if a Series is used as the key, the series labels are
+        preserved with its values use as the key.
 
         Keys will be turned into arkouda arrays as needed.
 
         Returns
         -------
-
         A Series containing the values corresponding to the key.
         """
-        t = type(key)
         if isinstance(key, Series):
             # special case, keep the index values of the Series, and lookup the values
             labels = key.index
-            key = key.values
-            v = lookup(self.index.index, self.values, key)
+            val = key.values
+            v = lookup(self.index.index, self.values, val)
             return Series((labels, v))
```

Comment:
we don't need this to be so many lines
```python
v = lookup(self.index.index, self.values, key.values)
return Series((key.index, v))
```
also i prefer the explicit approach of initializing a `Series` (not relying on tuple case), so perhaps a cleaner way to do this is
```python
return Series(index = key.index, data = lookup(self.index.index, self.values, key.values))
```

Reply:
Done